modules/software-development-handson/exercises_cicd.md

```diff
@@ -4,6 +4,47 @@ type: exercise
 order: 4
 ---
 
-# Good Practices will save the day!
+# Make your automated testing framework use less energy
+
+## Testing is necessary
+A robust testing framework is an essential part of good software development practices. It is even more vital in a research environment since wrong or misleading scientific results may be caused by bad or insufficient testing. Testing aids in scientific reproducibility and reduces the amount of time wasted trying to go back to find which version still worked with a given feature.
```

Comment:
```suggestion
A robust testing framework is an essential part of good software development practices. It is even more vital in a research environment since wrong or misleading scientific results may be caused by bad or insufficient testing. Automated testing aids in scientific reproducibility and reduces the amount of time wasted trying to go back to find which version still worked with a given feature.
```